# O Head de Dados pediu para que você construísse uma tabela contendo as seguintes informações:
- País de origem (Brasil).
- País de destino.
- Quantidade em litros de vinho exportado (utilize: 1KG =1L).
- Valor em US$

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.ticker as ticker
import warnings

#Ignorar mensagens de avisos
warnings.filterwarnings("ignore")

In [2]:
#Comando para exibir todas colunas do arquivo
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

#Comando utilizado para carregar o arquivo e armazena-lo como um DataFrame do Pandas
#Um DataFrame do Pandas é como se fosse uma planilha do Excel, onde podemos tratar linhas e colunas.
pd.options.display.float_format = '{:.2f}'.format

In [3]:
# IMPORTANDO dados 
dados_espumante = pd.read_csv('/Users/lucas/pos-tech/challenge/wine_data/wine_info/ExpEspumantes.csv', sep=';', engine='python',encoding='UTF-8')

In [4]:
dados_espumante.head()

Id                 País  1970  1970.1  1971  1971.1  1972  1972.1  1973  \
0   1        África do Sul     0       0     0       0     0       0     0   
1   2             Alemanha     0       0     0       0     0       0     0   
2   3               Angola     0       0     0       0     0       0     0   
3   4    Antigua e Barbuda     0       0     0       0     0       0     0   
4   5  Antilhas Holandesas     0       0     0       0     0       0     0   

   1973.1  1974  1974.1  1975  1975.1  1976  1976.1  1977  1977.1  1978  \
0       0     0       0     0       0     0       0     0       0     0   
1       0     0       0     0       0     0       0     0       0     0   
2       0     0       0     0       0     0       0     0       0     0   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   1978.1  1979  1979.1  1980  1980.1  1981  1981.1  1982  1982.1  1983  \
0       0     0       0     0       0     0       0     0       0     0   
1       0     0       0     0       0     0       0     0       0     0   
2       0     0       0     0       0     0       0     0       0     0   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   1983.1  1984  1984.1  1985  1985.1  1986  1986.1  1987  1987.1  1988  \
0       0     0       0     0       0     0       0     0       0     0   
1       0     0       0     0       0     0       0     0       0     0   
2       0     0       0     0       0     0       0     0       0     0   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   1988.1  1989  1989.1  1990  1990.1  1991  1991.1  1992  1992.1  1993  \
0       0     0       0     0       0     0       0     0       0     0   
1       0     0       0     0       0     0       0     0       0     0   
2       0     0       0     0       0     0       0     0       0     0   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   1993.1  1994  1994.1  1995  1995.1  1996  1996.1  1997  1997.1  1998  \
0       0     0       0     0       0     0       0     0       0     0   
1       0     0       0     0       0     0       0     0       0     0   
2       0   600    6276     0       0     0       0     0       0   430   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   1998.1   1999  1999.1  2000  2000.1  2001  2001.1  2002  2002.1  2003  \
0       0      0       0     0       0     0       0     0       0     0   
1       0      0       0     0       0  2816    5882     0       0     0   
2     491  19474   18696  1798   13018   784     492     0       0   368   
3       0      0       0     0       0     0       0     0       0     0   
4       0      0       0     0       0     0       0     0       0     0   

   2003.1  2004  2004.1  2005  2005.1  2006  2006.1  2007  2007.1  2008  \
0       0     0       0     0       0     0       0     0       0     0   
1       0   730    3045   549    2670   153     928  3547   10192  5423   
2     367   793    1040   419     656  3314    6657  6293   26252  2244   
3       0     0       0     0       0     0       0     0       0     0   
4       0     0       0     0       0     0       0     0       0     0   

   2008.1   2009  2009.1    2010  2010.1  2011  2011.1  2012  2012.1  2013  \
0       0      0       0       0       0     0       0     0       0     0   
1   20928   6648   29018    1958    6877  1976   12018   833    5072  5809   
2   10581  11393   33664  104138  672533   904   14196    73    1426  5850   
3       0      0       0       0       0     0       0     

## Sumário
- Id – dos dados inseridos
- País -> Coluna dos países para que o Brasil exportou
- 1970 e anos futuros – Valores exportados em quilogramas
- 1970.1 e anos futuros – Valores exportados em valor ($)

Perguntas a serem respondidos

1 – Qual a receita em US$ e quantidade em quilograma total?

2 - Quais países top 5 exportadores de uva fresca?

3 – Qual ou quais anos foram as maiores exportações e por quê?

4 – Qual ou quais anos foram as menores exportações e por quê?

5 – Qual é o preço médio por Quilo (USD/Kg)?

6 – Quais são as sazonalidades das maiores exportações?

7 – Quais dados externos influenciam nas exportações?

In [5]:
dados_espumante.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Columns: 110 entries, Id to 2023.1
dtypes: int64(109), object(1)
memory usage: 88.6+ KB


In [6]:
print(f'Dados nulos nas colunas: {dados_espumante.isna().sum().sum()}')

Dados nulos nas colunas: 0


In [7]:
dados_espumante.drop(columns='Id', inplace=True)

In [8]:
dados_espumante.tail()

País  1970  1970.1  1971  1971.1  1972  1972.1  1973  1973.1  1974  \
98     Uruguai     0       0     0       0     0       0     0       0     0   
99     Vanuatu     0       0     0       0     0       0     0       0     0   
100  Venezuela     0       0     0       0     0       0     0       0     0   
101     Vietnã     0       0     0       0     0       0     0       0     0   
102  Outros(1)  5132    3208     0       0  1200     579   144     102     0   

     1974.1  1975  1975.1  1976  1976.1  1977  1977.1  1978  1978.1  1979  \
98        0     0       0     0       0     0       0     0       0     0   
99        0     0       0     0       0     0       0     0       0     0   
100       0     0       0     0       0     0       0     0       0     0   
101       0     0       0     0       0     0       0     0       0     0   
102       0     0       0  1544    2726     0       0   113     176  1040   

     1979.1  1980  1980.1  1981  1981.1   1982  1982.1   1983  1983.1  1984  \
98        0     0       0     0       0      0       0      0       0     0   
99        0     0       0     0       0      0       0      0       0     0   
100       0     0       0     0       0  13280   19278  19706   33201     0   
101       0     0       0     0       0      0       0      0       0     0   
102     509  1786    1439  1009    1425    219     490     60     125     0   

     1984.1  1985  1985.1  1986  1986.1  1987  1987.1   1988  1988.1  1989  \
98        0     0       0     0       0     0       0      0       0     0   
99        0     0       0     0       0     0       0      0       0     0   
100       0     0       0     0       0     0       0      0       0     0   
101       0     0       0     0       0     0       0      0       0     0   
102       0  3108    3822   960    6223    36     299  14538   48908  5112   

     1989.1  1990  1990.1   1991  1991.1  1992  1992.1  1993  1993.1  1994  \
98        0     0       0      0       0     0       0     0       0     0   
99        0     0       0      0       0     0       0     0       0     0   
100       0     0       0      0       0     0       0     0       0     0   
101       0     0       0      0       0     0       0     0       0     0   
102    3075     0       0  35306   25786   628    1606  1227    2862     0   

     1994.1  1995  1995.1  1996  1996.1  1997  1997.1  1998  1998.1  1999  \
98        0    30      21   240     827    13      50   396     763  1814   
99        0     0       0     0       0     0       0     0       0     0   
100       0     0       0     0       0     0       0     0       0     0   
101       0     0       0     0       0     0       0     0       0     0   
102       0     0       0     0       0     0       0    42      89     0   

     1999.1  2000  2000.1  2001  2001.1  2002  2002.1  2003  2003.1  2004  \
98     1096  5040    3335  1212     639   176    1165     0       0   143   
99        0     0       0     0       0     0       0     0       0     0   
100       0     0       0     0       0     0       0     0       0     0   
101       0     0       0     0       0     0       0     0       0     0   
102       0     0       0   974    4895   494    2127     0       0     0   

     2004.1  2005  2005.1  2006  2006.1  2007  2007.1  2008  2008.1  2009  \
98      143     0       0  1075    2862     0       0     0       0     0   
99        0     0       0     0       0     0       0     0       0     0   
100       0     0       0     0       0     0       0     0       0     0   
101       0     0       0     0       0     0       0     0       0   675   
102       0     0       0     0       0  3082   13359     0       0     0   

     2009.1  2010  2010.1  2011  2011.1  2012  2012.1   2013  2013.1   2014  \
98        0   472   21325     0       0     0       0  22717   82754  34463   
99        0     0       0     0       0     0       0      0       0      0   
100       0     0       0  

In [9]:
#renomeando paises
from unidecode import unidecode

nome_dos_paises = {
        'Alemanha, Republica Democratica': 'Alemanha',
        'Alemanha, Republica Democratica da': 'Alemanha',
        'Belice': 'Belize',
        'Belize': 'Belize',
        'Camaroes': 'Camaroes',
        'Camores': 'Camaroes',
        'Coreia do Sul, Republica da': 'Coreia do Sul',
        'Coreia do Sul, Republica': 'Coreia do Sul',
        'Coreia do Sul': 'Coreia do Sul',
        'Coreia, Republica Sul': 'Coreia do Sul',
        'Coreia, Republica Sul': 'Coreia do Sul',
        'Dominica': 'Dominica',
        'Dominica, Ilha de': 'Dominica',
        'Eslovaca, Republica': 'Eslovaquia',
        'Falkland (Ilhas Malvinas)': 'Ilhas Malvinas',
        'Falkland (Malvinas)': 'Ilhas Malvinas',
        'Filanldia': 'Finlandia',
        'Finlandia': 'Finlandia',
        'Guine Bissau': 'Guine-Bissau',
        'Guine Bissau': 'Guine-Bissau',
        'Cayman, Ilhas': 'Ilhas Cayman',
        'Marshall, Ilhas': 'Ilhas Marshall',
        'Paises Baixos': 'Holanda',
        'Paises Baixos (Holanda)': 'Holanda',
        'Russia': 'Russia',
        'Russia, Federacao da': 'Russia',
        'Republica Federativa da Russia': 'Russia',
        'Taiwan (FORMOSA)': 'Taiwan',
        'Taiwan (Formosa)': 'Taiwan',
        'Trindade e Tobago': 'Trindade e Tobago',
        'Trinidade Tobago': 'Trindade e Tobago',
        'Trinidade e Tobago': 'Trindade e Tobago',
        'Tcheca, Republica': 'Republica Tcheca',
        'Georgia do Sul e Sandwich do Sul, Ilhas': 'Ilhas Georgia do Sul e Sandwich do Sul'
}

dados_espumante = dados_espumante.rename(nome_dos_paises)

In [10]:
dados_espumante.set_index('País', inplace=True)

In [11]:
def separar_valor_kg(dados):
    coluna_valor = dados.columns[dados.columns.str.contains('\.')]
    coluna_kg = dados.columns[~dados.columns.str.contains('\.')]
    dados_espumante_valor = dados[coluna_valor]
    dados_espumante_valor.index = dados.index
    dados_espumante_kg = dados[coluna_kg]
    return dados_espumante_valor, dados_espumante_kg



In [12]:
df_espumante_valor, df_espumante_kg = separar_valor_kg(dados_espumante)

In [13]:
display(df_espumante_kg.tail(2))
display(df_espumante_valor.tail(2))

1970  1971  1972  1973  1974  1975  1976  1977  1978  1979  1980  \
País                                                                          
Vietnã        0     0     0     0     0     0     0     0     0     0     0   
Outros(1)  5132     0  1200   144     0     0  1544     0   113  1040  1786   

           1981  1982  1983  1984  1985  1986  1987   1988  1989  1990   1991  \
País                                                                            
Vietnã        0     0     0     0     0     0     0      0     0     0      0   
Outros(1)  1009   219    60     0  3108   960    36  14538  5112     0  35306   

           1992  1993  1994  1995  1996  1997  1998  1999  2000  2001  2002  \
País                                                                          
Vietnã        0     0     0     0     0     0     0     0     0     0     0   
Outros(1)   628  1227     0     0     0     0    42     0     0   974   494   

           2003  2004  2005  2006  2007  2008  2009  2010  2011  2012  2013  \
País                                                                          
Vietnã        0     0     0     0     0     0   675   189     0     0     0   
Outros(1)     0     0     0     0  3082     0     0     0     0     0     0   

           2014  2015  2016  2017  2018  2019  2020  2021  2022  2023  
País                                                                   
Vietnã        0     0     0     0     0     0     0   144    16     0  
Outros(1)     0     0     0     0     0     0     0     0     0     0

1970.1  1971.1  1972.1  1973.1  1974.1  1975.1  1976.1  1977.1  \
País                                                                        
Vietnã          0       0       0       0       0       0       0       0   
Outros(1)    3208       0     579     102       0       0    2726       0   

           1978.1  1979.1  1980.1  1981.1  1982.1  1983.1  1984.1  1985.1  \
País                                                                        
Vietnã          0       0       0       0       0       0       0       0   
Outros(1)     176     509    1439    1425     490     125       0    3822   

           1986.1  1987.1  1988.1  1989.1  1990.1  1991.1  1992.1  1993.1  \
País                                                                        
Vietnã          0       0       0       0       0       0       0       0   
Outros(1)    6223     299   48908    3075       0   25786    1606    2862   

           1994.1  1995.1  1996.1  1997.1  1998.1  1999.1  2000.1  2001.1  \
País                                                                        
Vietnã          0       0       0       0       0       0       0       0   
Outros(1)       0       0       0       0      89       0       0    4895   

           2002.1  2003.1  2004.1  2005.1  2006.1  2007.1  2008.1  2009.1  \
País                                                                        
Vietnã          0       0       0       0       0       0       0    1768   
Outros(1)    2127       0       0       0       0   13359       0       0   

           2010.1  2011.1  2012.1  2013.1  2014.1  2015.1  2016.1  2017.1  \
País                                                                        
Vietnã        651       0       0       0       0       0       0       0   
Outros(1)       0       0       0       0       0       0       0       0   

           2018.1  2019.1  2020.1  2021.1  2022.1  2023.1  
País                                                       
Vietnã          0       0       0     374      19       0  
Outros(1)       0       0       0       0       0       0

In [14]:
df_espumante_kg.drop(df_espumante_kg.index[-1],axis=0, inplace=True)
df_espumante_valor.drop(df_espumante_valor.index[-1],axis=0, inplace=True)

In [15]:
def dados_aberto(dados, label = 'coluna'):
    if dados.index is 'País':
        dados = dados.set_index('País').T.reset_index()
        dados_aberto = pd.melt(dados, id_vars=['index'],var_name='pais', value_name=f'{label}')
    else:
        dados = dados.T.reset_index()
        dados_aberto = pd.melt(dados, id_vars=['index'],var_name='pais', value_name=f'{label}')
    return dados_aberto

In [16]:
df_espumante_valor_aberto = dados_aberto(df_espumante_valor, 'us$')
df_espumante_valor_aberto.rename(columns={'index':'ano'}, inplace=True)


In [17]:
df_espumante_kg_aberto = dados_aberto(df_espumante_kg, 'kg')
df_espumante_kg_aberto.rename(columns={'index':'ano'}, inplace=True)
df_espumante_kg_aberto.head()

ano           pais  kg
0  1970  África do Sul   0
1  1971  África do Sul   0
2  1972  África do Sul   0
3  1973  África do Sul   0
4  1974  África do Sul   0

tratando as linhas da coluna ano

In [18]:
df_espumante_valor_aberto['ano'] = df_espumante_valor_aberto['ano'].map(lambda x: str(x).replace('.1',''))


In [19]:
df_espumante_valor_aberto.head()


ano           pais  us$
0  1970  África do Sul    0
1  1971  África do Sul    0
2  1972  África do Sul    0
3  1973  África do Sul    0
4  1974  África do Sul    0

# Top 5 paises em valor e kg

In [20]:
df_espumante_kg_top5 = df_espumante_kg_aberto.query('ano >= "2009" and ano <= "2023"').drop(columns='ano').groupby('pais').sum().sort_values('kg', ascending=False).head(5)

In [21]:
df_espumante_valor_top5 = df_espumante_valor_aberto.query('ano >= "2009" and ano <= "2023"').drop(columns='ano').groupby('pais').sum().sort_values('us$', ascending=False).head(5)

In [22]:
display(df_espumante_valor_top5)
display(df_espumante_kg_top5)

us$
pais                   
Estados Unidos  9788986
Paraguai        1680233
Reino Unido     1589727
Angola          1226906
Belgica          624956

kg
pais                   
Estados Unidos  3388562
Paraguai         596129
Reino Unido      339837
Angola           213033
China            166101

In [23]:
df_espumante = pd.merge(df_espumante_kg_aberto[['ano','pais','kg']], df_espumante_valor_aberto['us$'], left_index= True,
                        right_index=True, how='inner')

In [24]:
df_espumante['ano'] = df_espumante['ano'].astype(int)
df_espumante.head()

ano           pais  kg  us$
0  1970  África do Sul   0    0
1  1971  África do Sul   0    0
2  1972  África do Sul   0    0
3  1973  África do Sul   0    0
4  1974  África do Sul   0    0

In [25]:
import numpy as np

df_espumante = df_espumante.query('ano >= 2008')


In [26]:
df_espumante.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1632 entries, 38 to 5507
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   ano     1632 non-null   int64 
 1   pais    1632 non-null   object
 2   kg      1632 non-null   int64 
 3   us$     1632 non-null   int64 
dtypes: int64(3), object(1)
memory usage: 63.8+ KB


In [27]:
df_espumante['us$/kg'] = df_espumante.apply(lambda row: 0 if row['kg'] == 0 else row['us$']/row['kg'], axis=1)
#df_espumante['us$/kg'] = np.where(df_espumante['kg'] != 0, df_espumante['us$'] / df_espumante['kg'], 0)


In [28]:
df_espumante.set_index('ano', inplace=True)

In [29]:
#df_espumante.to_csv('/Users/lucas/pos-tech/challenge/wine_data/dados_tratados/df_espumante_aberto.csv')